open-sphere-plugins/arcgis/src/main/java/io/opensphere/arcgis2/envoy/ArcGISLayerListEnvoy.java

```diff
@@ -166,20 +239,53 @@ public void query(DataModelCategory category, Collection<? extends Satisfaction>
         }
     }
 
+    /**
+     * Reads the response from the supplied URL from the supplied input stream.
+     * If debug is enabled, the response is logged before parsing.
+     *
+     * @param url the URL from which the response was received.
+     * @param is the input stream to read.
+     * @return a {@link FolderInfo} object containing the unmarshalled JSON
+     *         response.
+     * @throws IOException if the input stream cannot be read.
+     * @throws JsonParseException if the contents of the input stream cannot be
+     *             processed as JSON data.
+     * @throws JsonMappingException if the contents of the input stream cannot
+     *             be processed as JSON data.
+     */
+    private FolderInfo parseResponse(URL url, InputStream is) throws IOException, JsonParseException, JsonMappingException
+    {
+        FolderInfo info;
+        if (LOGGER.isDebugEnabled())
+        {
+            Charset charset = Charset.forName(System.getProperty("opensphere.charset", "UTF-8"));
```

Comment:
Can you replace this with StringUtilities.DEFAULT_CHARSET?

Reply:
I was wondering where that was. Fixed.